metabase/metabase
# Hide dashboard filters if they're not mapped to dash cards on the current tab

Resolves #36359

Creators currently have to deal with the limitation of having filters show up in all dashboard contexts all the time, regardless if they are relevant to the charts. We already have tabs as a way to split up your dashboard, so we’ll leverage that and simply provide a way to only show filters that are relevant to certain tabs.

This PR will hide dashboard filters on tabs where they're not connected to at least one card. OTOH, in the editing mode, we show the full list on every tab.

> [!NOTE]  
> A handful of E2E tests started failing because some filters are now hidden. Fixes for these tests are actually the biggest chunk of the diff here

### To verify

1. Create a dashboard with at least two tabs, a few filters, and some questions on it
2. Map your dashboard filters, so you have
	- a filter connected to at least one card on every tab
	- a filter connected to cards only on one tab
	- a filter not connected to any cards at all
3. Ensure a filter gets hidden on tabs where they're not connected to any cards
4. Ensure all filters are visible on every tab in dashboard editing mode

### Demo

https://github.com/metabase/metabase/assets/17258145/0182de9e-85e8-479b-8bab-cbeaf703b076
